core/modules/filters/interleave.js

```diff
@@ -0,0 +1,75 @@
+/*\
+title: $:/core/modules/filters/interleave.js
+type: application/javascript
+module-type: filteroperator
+
+Interleave two or more lists one item at a time
+
+\*/
+(function(){
+
+	/*jslint node: true, browser: true */
+	/*global $tw: false */
+	"use strict";
+
+	/*
+	Interleave two or more lists
+	*/
+	exports.interleave = function(source,operator,options) {
+		var allowDuplicates = false;
+		switch(operator.suffix) {
+			case "raw":
+				allowDuplicates = true;
+				break;
+			case "dedupe":
+				allowDuplicates = false;
+				break;
+		}
+		var results = new $tw.utils.LinkedList();
```

Comment:
@flibbles ... Does the linked-list have an advantage here?

Reply:
The advantage of the linked list here is that I can call `results.makeTiddlerIterator` and there won't be an array created and thrown away, which reduces TW's memory usage. It's unlikely to make much difference since `interleave` isn't likely to be called with large numbers of items in most usage scenarios, but every little bit can help.

Reply:
It is advantageous here, but I've got a concern I'll put in the main thread about the deduping.